### PR TITLE
Gozag potion petition improvements

### PR DESCRIPTION
**Pull request by alexjurkiewicz**
Note: This branch is not yet ready for merge. I'm opening the pull request for feedback.

### CODE REVIEW

File: crawl-ref/source/godabil.cc [hunk 1]
```diff
@@ -4242,6 +4242,49 @@ bool gozag_setup_potion_petition(bool quiet)
     return true;
 }
 
+// Would a certain potion effect be immediately useful?
+// TODO: check duration for all buff-extending effects (might, haste, etc)
+int _potion_effect_pricing(item_def potion)
+{
+    int price = item_value(potion, true);
+    switch (potion.sub_type)
```
Comment (alexjurkiewicz) [inline crawl-ref/source/godabil.cc]: This switch statement is too big. I'd like to reduce the price of any potion if `!get_potion_effect(potion.sub_type)->can_quaff()` but that formulation gives me a type casting error. Not sure what I'm doing wrong.
Once that's done, only special cases will belong in this select statement (like flight & berserk).

File: crawl-ref/source/godabil.cc [hunk 2]
```diff
@@ -4259,7 +4302,7 @@ bool gozag_potion_petition()
             for (int i = 0; i < GOZAG_MAX_POTIONS; i++)
             {
                 prices[i] = 0;
-                int multiplier = 25; // arbitrary
+                int multiplier = 20 + random2(10); // arbitrary
```
Comment (alexjurkiewicz) [inline crawl-ref/source/godabil.cc]: With the change to reduce prices of less-useful potions, this maybe should increase.

File: crawl-ref/source/potion.cc [hunk 3]
```diff
@@ -73,7 +69,11 @@ class PotionCuring : public PotionEffect
             return true;
         }
 
-        return false;
```
Comment (alexjurkiewicz) [inline crawl-ref/source/potion.cc]: I'm really not 100% about this logic here. Can someone double check??

Comment (amalloy) [inline crawl-ref/source/potion.cc]: I think this makes it impossible for players to quaff a heal_wounds manually when they're at full HP and not rotted. Maybe that's a good change, but I'm not sure it's a good idea to sneak that in with a gozag change.

Comment (alexjurkiewicz) [inline crawl-ref/source/potion.cc]: Fair call, I should have mentioned this in the issue body.

File: crawl-ref/source/godabil.cc [hunk 4]
```diff
@@ -4242,48 +4244,85 @@ bool gozag_setup_potion_petition(bool quiet)
     return true;
 }
 
-bool gozag_potion_petition()
+// What's the price of a particular potion effect?
+int _gozag_price_effect(item_def potion)
```
Comment (alexjurkiewicz) [inline crawl-ref/source/godabil.cc]: godabil.cc: In function ‘int _gozag_price_effect(item_def)’:
    godabil.cc:4248:5: warning: no previous declaration for ‘int _gozag_price_effect(item_def)’ [-Wmissing-declarations]
 int _gozag_price_effect(item_def potion)

Not sure how to fix this

Comment (amalloy) [inline crawl-ref/source/godabil.cc]: @alexjurkiewicz I think you just need to add a `static` to the front: `static int ...`

**Comment by MarvinPA** [comment]
I don't see any reason to guarantee healing and magic, seems very unnecessary to me.

**Comment by amalloy** [comment]
I agree @MarvinPA. In my Mu^Gozag game, I was usually not looking for healing potions anyway, but for buffs. And if I did need a heal, well, most petitions included a heal anyway.

**Comment by MarvinPA** [comment]
I'm not sure about the faith removal thing but I think a one-time cost increase for all three abilities would be better than just for potion petition, maybe.

**Comment by alexjurkiewicz** [comment]
I've removed guaranteed healing/magic. Do you still think faith should increase the price for all abilities? I'll update the commit if so.

**Comment by alexjurkiewicz** [comment]
@MarvinPA I think this is ready for merge -- any comments if not?

**Comment by MarvinPA** [comment]
I'm not sure about discounting extended potions - I think having a good effect extended is still pretty useful. I don't really know about discounting useless ones either for that matter, it seems possibly just like it makes the ability unnecessarily more complex for not a huge gain.

I definitely like tying Gozag gold drops to something other than monster size, threat is probably reasonable there.

Also still unsure about faith generally, seems tricky to give it a reasonably consistent and intuitive effect on removal - it might actually be more reasonable to just have faith have no effect with Gozag, since Gozag doesn't care about anything other than gold itself.

**Comment by alexjurkiewicz** [comment]
I've removed the commit that changes gold drop values since it needs further tweaking before it's ready for commit.

**Comment by MarvinPA** [comment]
Pushed the variable potion pricing, thanks. Having gold drops work in some different way would be nice but that probably needs more thought and can be a separate thing.

**Pull request closed without merging**